src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -1951,20 +1953,26 @@ static Set<String> expandFileNames(String[] filePathList, String queryId)
         // The following currently ignore sub directories
         for (Object file :
             FileUtils.listFiles(dir, new WildcardFileFilter(entry.getValue()), null)) {
-          result.add(((java.io.File) file).getCanonicalPath());
+          result.add(((File) file).getCanonicalPath());
         }
       } catch (Exception ex) {
-        throw new SnowflakeSQLException(
-            queryId,
-            ex,
-            SqlState.DATA_EXCEPTION,
-            ErrorCode.FAIL_LIST_FILES.getMessageCode(),
-            "Exception: "
-                + ex.getMessage()
-                + ", Dir="
-                + entry.getKey()
-                + ", Patterns="
-                + entry.getValue().toString());
+        if (ex instanceof UncheckedIOException && dir != null && !dir.exists()) {
+          logger.warn(
```

Comment:
Just to be on the safe side - if dir is null, we just swallow the exception without any log or anything, maybe it is worth adding some logs?